ONSdigital/response-operations-ui
# SUS005 Change Social Case Status

# Motivation and Context

SEL users need to be able to change the status of a social case to match the outcomes of phone calls with respondents. This PR adds a spoke page to change the status to anything from a list of available allowed events.

# What has changed



* Added spoke to select new status
* Added change status button to social case details page
* Added mappings for allowed statuses and events
* Added long description if current case status is a social outcome status

# How to test?



* Run this branch with the other SUS005 service branches
* Execute a social collection exercise and navigate to the case details for a postcode in that collection exercise
* When the current status is `Not Started`, `In Progress` or `Complete` there should the be the option to change the case status which opens a spoke page allowing you to select an outcome from a list
* When you select an outcome and confirm you should be taken back to the case details with a success banner and the correct new status displayed with description of the outcome event
# Links



https://trello.com/c/2jyQfOjw/191-sus005-change-the-status-of-an-lms-case-13#

Depends on these branches:
https://github.com/ONSdigital/rm-case-service/pull/84
https://github.com/ONSdigital/rm-casesvc-api/pull/29
